code/modules/power/tesla/energy_ball.dm

```diff
@@ -0,0 +1,157 @@
+			loc = get_step(src,move_dir)
+
+/obj/singularity/energy_ball/proc/handle_energy()
+	if(energy >= 300)
+		energy = 0
+		var/obj/singularity/energy_ball/EB = new(loc)
+		orbiting_balls.Add(EB)
+		EB.transform *= 0.5
+		EB.is_orbiting = 1
+		var/icon/I = icon(icon,icon_state,dir)
+
+		var/orbitsize = (I.Width()+I.Height())*0.5
+		orbitsize -= (orbitsize/world.icon_size)*(world.icon_size*0.25)
+		spawn(1)
+			EB.orbit(src,orbitsize,0)
+
+
+
+/obj/singularity/energy_ball/Bump(atom/A)
+	if(istype(A, /mob/living/carbon))
+		var/mob/living/carbon/C = A
+		C.dust()
+	return
+
+/obj/singularity/energy_ball/Bumped(atom/A)
+	if(istype(A, /mob/living/carbon))
+		var/mob/living/carbon/C = A
+		C.dust()
+	return
+
+proc/get_closest_atom(var/type, var/list, var/source)
+	var/closest_atom
+	var/closest_distance
+	for(var/A in list)
+		if(!istype(A, type))
+			continue
+		var/distance = get_dist(source, A)
+		if(!closest_distance)
+			closest_distance = distance
+			closest_atom = A
+		else
+			if(closest_distance > distance)
+				closest_distance = distance
+				closest_atom = A
+	return closest_atom
+
+proc/tesla_zap(var/atom/source, var/zap_range = 3, var/power)
+	if(power < 1000)
+		return
+	var/list/tesla_coils = list()
+	var/list/potential_machine_zaps = list()
+	var/list/potential_mob_zaps = list()
+	var/closest_atom
+	for(var/atom/A in orange(source, zap_range))
+		if(istype(A, /obj/machinery/power/tesla_coil))
+			var/obj/machinery/power/tesla_coil/C = A
+			if(C.being_shocked)
+				continue
+			tesla_coils.Add(C)
+			continue
+		if(istype(A, /obj/machinery))
+			var/obj/machinery/M = A
+			if(is_type_in_list(M, blacklisted_tesla_types))
+				continue
+			if(M.being_shocked)
+				continue
+			potential_machine_zaps.Add(M)
+			continue
+		if(istype(A, /mob/living))
+			var/mob/living/L = A
+			if(L.stat == DEAD)
+				continue
+			if(is_type_in_list(L, blacklisted_tesla_types))
+				continue
+			potential_mob_zaps.Add(L)
+			continue
+	closest_atom = get_closest_atom(/obj/machinery/power/tesla_coil, tesla_coils, source)
+	if(closest_atom && istype(closest_atom, /obj/machinery/power/tesla_coil))
+		var/obj/machinery/power/tesla_coil/C = closest_atom
+		source.Beam(C,icon_state="lightning",icon='icons/effects/effects.dmi',time=5)
+		C.tesla_act(power)
+		return
+	if(!closest_atom)
+		closest_atom = get_closest_atom(/mob/living, potential_mob_zaps, source)
+		if(closest_atom && istype(closest_atom, /mob/living))
+			var/mob/living/L = closest_atom
+			var/shock_damage = Clamp(round(power/400), 10, 200) + rand(-5,5)
+			source.Beam(L,icon_state="lightning",icon='icons/effects/effects.dmi',time=5)
+			L.electrocute_act(shock_damage, source, 1)
+			return
+	if(!closest_atom)
+		closest_atom = get_closest_atom(/obj/machinery, potential_machine_zaps, source)
+		if(closest_atom)
+			var/obj/machinery/M = closest_atom
+			source.Beam(M,icon_state="lightning",icon='icons/effects/effects.dmi',time=5)
+			M.tesla_act(power)
+			if(prob(85))
+				M.emp_act(3)
```

Comment:
Severity 3 on emp_act does nothing on most things, as emp_act expects a severity of 1 or 2